main.go

```diff
@@ -118,50 +119,61 @@ func publishSpec(vargs API) error {
 		return errors.Wrap(err, "unable to close multipart payload")
 	}
 
-	// grabbing body in case we need to retry
+	var success bool
 	payload := body.Bytes()
 	contentType := w.FormDataContentType()
-	var success bool
 	// make request with timeouts & retries
 	for attempt := 1; attempt < 4; attempt++ {
 		fmt.Printf("attempting to publish spec file: %s\n", vargs.Spec)
-		r, err := http.NewRequest(http.MethodPost, vargs.UploaderURL+"?key="+vargs.Key,
-			bytes.NewBuffer(payload))
-		if err != nil {
-			return errors.Wrap(err, "unable to create request")
-		}
-		r.Header.Set("Content-Type", contentType)
-		resp, err := makeRequest(r)
-		if err == nil && resp != nil && resp.StatusCode == http.StatusOK {
-			resp.Body.Close()
+		status, resp, err := makeRequest(
+			vargs.UploaderURL+"?key="+vargs.Key,
+			contentType,
+			payload)
+		if err == nil && status == http.StatusOK {
 			success = true
 			break
 		}
-
 		if err != nil {
-			fmt.Printf("problems publishing spec on attempt %d: %s\nsleeping for 1s\n", attempt, err)
+			fmt.Printf("problems publishing spec on attempt %d: %s\n", attempt, err)
 		} else {
-			respBod, _ := ioutil.ReadAll(resp.Body)
-			resp.Body.Close()
-			fmt.Printf("problems publishing spec on attempt %d: uploader returned with bad status of %d - %s\nsleeping for 1s\n",
-				attempt, resp.StatusCode, string(respBod))
+			fmt.Printf("problems publishing spec on attempt %d: %d - %s\n",
+				attempt, status, resp)
 		}
 		if attempt < 3 {
-			time.Sleep(1 * time.Second)
+			dur := time.Duration(attempt) * time.Second
+			fmt.Printf("sleeping for %s\n", dur)
+			time.Sleep(dur)
 		}
 	}
 	if success {
-		fmt.Printf("successfully published spec file: %s\n", vargs.Spec)
+		fmt.Println("successfully published spec file")
 		return nil
 	}
 	return errors.New("unable to publish specs after 3 attempts")
 }
 
-func makeRequest(r *http.Request) (*http.Response, error) {
+func makeRequest(url, contentType string, payload []byte) (int, []byte, error) {
+	r, err := http.NewRequest(http.MethodPost, url,
+		bytes.NewBuffer(payload))
```

Comment:
Why split this on a new line?

Reply:
It was longer at first (>90 char), I just hadn't moved it back.

I generally have my vim setup to highlight code differently once it hits that length as a reminder so I start wrapping lines.

```
colorscheme ron
let &colorcolumn=join(range(91,999),",")
highlight ColorColumn ctermbg=235 guibg=#2c2d27
```